### PR TITLE
Code refactoring, stats logger

### PR DESCRIPTION
**Pull request by mprtcz**
Added fields for datetime and bookstore name to db model
Added tests to Shop enum
Fixed toString test for BookModel
BookSender refactoring. Common code for scrappers. 
Merging BookSender refactoring. Common code for scrappers. 
Merge remote-tracking branch 'origin/development' into development 			
Added logging of times and books gathered for the remaining scrappers;
There is now no error when the GoogleAPI is unavailable;…
Refactored Bonito scrapper to have the same methods as other scrapper
Refactoring code smells.
Removed some logging when parsing.
Merging refactored Bonito scrapper, fixed code smells
Merge remote-tracking branch 'origin/development' into development 		
Extracted duplicates to abstract base class
Merging: extracting duplicates to abstract base class
Added method for logging and removed duplications in scrapper main cl…
The App class is not much less bloated -> increased abstraction;
Removed sout from GoogleScrapper; 		
Added test for BookSender;
Removed obsolete souts;
Merge branch 'development' into scrapper-front
Merge refactored the scrapper code
Added code removed by merge to development;

### CODE REVIEW

File: crawler/src/main/java/com/javaacademy/crawler/App.java [hunk 1]
```diff
@@ -13,57 +11,49 @@
...
-        GoogleScrapper googleScrapper = new GoogleScrapper();
-        googleScrapper.runScrapping();
-
-        while (!googleScrapper.areAllCallbacksDone()) {
-            sleepFor(6000, "");
+    void runApp(Set<Scrapper> scrappers) {
+        for (Scrapper scrapper : scrappers) {
+            Set<BookModel> books = scrapper.scrape();
+            sendScrappedBooks(books, scrapper.getName());
         }
```
Comment (Oplia512) [inline crawler/src/main/java/com/javaacademy/crawler/App.java]: you can use java 8 feature scrappers.forEach( scrapper -> {Set<BookModel> books = scrapper.scrape(); 
sendScrappedBooks(books, scrapper.getName()})

File: crawler/src/main/java/com/javaacademy/crawler/App.java [hunk 2]
```diff
@@ -13,57 +11,49 @@
...
+    }
+
+    private void sendScrappedBooks(Set<BookModel> scrappedBooks, String bookStoreName) {
+        BookSender bookSender = new BookSender(scrappedBooks);
+        sendBookSAndLogInfo(bookSender, bookStoreName);
```
Comment (Oplia512) [inline crawler/src/main/java/com/javaacademy/crawler/App.java]: sendBookSAndLogInfo typo

File: crawler/src/main/java/com/javaacademy/crawler/common/booksender/BookSender.java [hunk 3]
```diff
@@ -11,31 +9,22 @@
...
-import java.util.concurrent.TimeUnit;
 import java.util.function.Consumer;
 
 import static com.javaacademy.crawler.common.logger.AppLogger.*;
 import static com.javaacademy.crawler.common.util.CrawlerUtils.sleepFor;
 
 public class BookSender {
 
-    Map<BookModel, Boolean> booksToSend; // a map which holds info about whether the book was successfully sent to server
+    Map<BookModel, Boolean> booksToSend; // holds info about whether the book was successfully sent to server
```
Comment (Oplia512) [inline crawler/src/main/java/com/javaacademy/crawler/common/booksender/BookSender.java]: make it final

File: crawler/src/main/java/com/javaacademy/crawler/common/booksender/BookSender.java [hunk 4]
```diff
@@ -11,31 +9,22 @@
...
-        bookModels.forEach(bookItem ->
-                booksToSend.put(bookItem, false));
-    }
 
     public BookSender(Set<BookModel> books) {
-        this.booksToSend = new HashMap<>();
-        this.booksSendCounter = new HashMap<>();
+        booksToSend = new HashMap<>();
+        booksSendCounter = new HashMap<>();
         books.forEach(bookItem -> booksToSend.put(bookItem, false));
```
Comment (Oplia512) [inline crawler/src/main/java/com/javaacademy/crawler/common/booksender/BookSender.java]: +1

File: crawler/src/main/java/com/javaacademy/crawler/App.java (hunk 1 above)
Comment (mprtcz) [reply crawler/src/main/java/com/javaacademy/crawler/App.java]: :+1:

File: crawler/src/main/java/com/javaacademy/crawler/googlebooks/GoogleScrapper.java [hunk 5]
```diff
@@ -64,7 +72,7 @@ void collectAllBooksFromGoogle(int numOfBooks) {
             getGoogleBooks(i, end, bookItemBookAddingCallback);
             sleepFor(sleepTime, "");
         }
-        displayProgress(100);
+        displayProgress(100, 100);
```
Comment (Oplia512) [inline crawler/src/main/java/com/javaacademy/crawler/googlebooks/GoogleScrapper.java]: magic numbers

File: crawler/src/main/java/com/javaacademy/crawler/googlebooks/model/GoogleBooksWrapper.java [hunk 6]
```diff
@@ -20,6 +21,7 @@
 
     @Override
     public List<Book> getItems() {
+        if(items == null) {return new ArrayList<>();}
```
Comment (Oplia512) [inline crawler/src/main/java/com/javaacademy/crawler/googlebooks/model/GoogleBooksWrapper.java]: one million dollar mistake ;) use Collection.EMPTY_LIST instead null

File: crawler/src/main/java/com/javaacademy/crawler/jsoup/JsoupBookScrapper.java [hunk 7]
```diff
@@ -1,23 +1,54 @@
...
+            return new Random().nextLong();
+        }
+        return isbn;
+    }
+
+    static double parsePrice(String s) {
+        double price;
+        try {
+            price = Double.parseDouble(s);
+        } catch (NumberFormatException e) {
```
Comment (Oplia512) [inline crawler/src/main/java/com/javaacademy/crawler/jsoup/JsoupBookScrapper.java]: log it

**Pull request merged**